src/main/java/pl/cleankod/exchange/provider/CurrencyConversionNbpService.java

```diff
@@ -17,10 +18,14 @@ public CurrencyConversionNbpService(ExchangeRatesNbpClient exchangeRatesNbpClien
     }
 
     @Override
+    @Cacheable(
```

Comment:
This is evil. Sooner or later you will lose control over that. IMO it would be better to use it explicitly via dependency.

Reply:
How would you like to resolve this? Is EhCache with explicit call in this methods is good enough? Or you have some other suggestions about this?